typescript/phoenix-sdk/examples/placeLimitOrder.ts

```diff
@@ -0,0 +1,120 @@
+  const phoenix = await Phoenix.Client.create(
+    connection,
+    endpoint,
+    traderKeypair.publicKey
+  );
+  let _airdropSig = await phoenix.connection.requestAirdrop(
+    traderKeypair.publicKey,
+    1_000_000_000
+  );
+
+  let marketAddress = new PublicKey(
+    "CS2H8nbAVVEUHWPF5extCSymqheQdkd4d7thik6eet9N"
+  );
+
+  let limitOrderPacket = Phoenix.getLimitOrderPacket(Phoenix.Side.Bid, 200, 1);
+  let marketData = phoenix.markets.get(marketAddress.toBase58())?.data;
+
+  if (marketData === undefined) {
+    throw Error("Market not found");
+  }
+
+  let limitOrderIxsNewTrader = await getLimitOrderNewMakerIxs(
+    connection,
+    marketAddress,
+    marketData,
+    traderKeypair.publicKey,
+    limitOrderPacket
+  );
+
+  let tx = new Transaction().add(...limitOrderIxsNewTrader);
+
+  await phoenix.connection.requestAirdrop(
+    traderKeypair.publicKey,
+    1_000_000_000
+  );
+  const txId = await phoenix.connection.sendTransaction(tx, [traderKeypair], {
+    skipPreflight: true,
+  });
+  await phoenix.connection.confirmTransaction(txId, "confirmed");
+  console.log("Place limit order Tx: ", txId);
+}
+
+export async function placeLimitOrder() {
+  // Enter keypair of trader with devnet tokens
+  // DO NOT USE KEY IN PRODUCTION
+  let secretKey = Buffer.from([
+    142, 233, 226, 117, 203, 164, 239, 67, 147, 205, 206, 191, 148, 4, 215, 24,
+    239, 154, 80, 211, 52, 212, 117, 142, 18, 70, 195, 148, 150, 109, 110, 20,
+    113, 47, 200, 185, 82, 117, 65, 107, 170, 218, 96, 124, 191, 124, 80, 129,
+    224, 175, 28, 89, 172, 136, 105, 156, 63, 70, 131, 151, 205, 77, 28, 236,
+  ]);
+  let traderKeypair = Keypair.fromSecretKey(secretKey);
+  let endpoint = "https://api.devnet.solana.com";
+  const connection = new Connection(endpoint);
+  const phoenix = await Phoenix.Client.create(
+    connection,
+    endpoint,
+    traderKeypair.publicKey
+  );
+
+  console.log("Trader pubkey: ", traderKeypair.publicKey.toBase58());
+  // let _airdropSig = await phoenix.connection.requestAirdrop(traderKeypair.publicKey, 1_000_000_000);
+
+  let marketAddress = new PublicKey(
+    "CS2H8nbAVVEUHWPF5extCSymqheQdkd4d7thik6eet9N"
+  );
+  let marketData = phoenix.markets.get(marketAddress.toBase58())?.data;
+
+  if (marketData === undefined) {
+    throw Error("Market not found");
+  }
+
+  // Claim seat ix
+  let claimSeatIx = getClaimSeatIx(marketAddress, traderKeypair.publicKey);
+  // place limit order ix
+  let limitOrderPacket = Phoenix.getLimitOrderPacket(Phoenix.Side.Bid, 200, 20);
+  let limitOrderIx = Phoenix.getPlaceLimitOrderSingleIx(
+    marketAddress,
+    marketData,
+    traderKeypair.publicKey,
+    limitOrderPacket
+  );
+  let tx = new Transaction().add(claimSeatIx, limitOrderIx);
+  const txId = await phoenix.connection.sendTransaction(tx, [traderKeypair], {
+    skipPreflight: true,
+  });
+  await phoenix.connection.confirmTransaction(txId, "confirmed");
+  console.log("Place limit order Tx: ", txId);
+}
```

Comment:
lots of duplicated code. why do these need to be two different methods? any downside to making it one long script like the rust one?

Reply:
The two functions serve as distinct examples in my head - the interaction if you're a new maker and the interaction if you were a maker previously. Do you think we should just combine into one?

Reply:
(Note that the new commit changes the name of the first example functions to make that distinction more clear)